h2/src/main/org/h2/server/pg/PgServerThread.java

```diff
@@ -582,6 +583,21 @@ private void writeDataColumn(JdbcResultSet rs, int column, int pgType, boolean t
                 write(data);
                 break;
             }
+            case PgServer.PG_TYPE_TEXTARRAY:
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                baos.write('{');
+                Value[] values = ((ValueArray) v).getList();
+                Charset encoding = getEncoding();
+                for (int i = 0; i < values.length; i++) {
+                    if (i > 0) {
+                        baos.write(',');
+                    }
+                    baos.write(values[i].getString().getBytes(encoding));
```

Comment:
I don't see any special treatment for `,` and `}` characters, but I guess it should be here.

Reply:
Sorry for my carelessness. Should quote for `",{}` and escape `"`.